### ISSUE TITLE
autolayer() seems to have troble with NaN

### ISSUE THREAD

**Issue by luiscarlosjr** [robjhyndman/forecast#619]
Found an issue trying to plot a forecast series containing NaN:
``` R
> plot_1 <- autoplot(trainning_series, series = "Data") + 
        autolayer(fc_1, series = "Forecast") +
        autolayer(fitted(fc_1), series = "Fitted")

#Error in if (leveldiff == 0) { : 
#  valeur manquante là où TRUE / FALSE est requis
```
On the other hand, this works:
``` R
> plot_1 <- autoplot(fc_1, series = "Forecast") +  # autoplot() is able to deal with the forecast object
        autolayer(trainning_series, series = "Data") +
        autolayer(fitted(fc_1), series = "Fitted")
```
The object `fc_1` was generated by forecasting a fit generated by `auto.arima()`.

I'm using the latest and greatest forecast package from github, R 3.3.1 linux 64-bit and ggplot 2.2.1.

Thank you for your attention and have a nice day :)

[autolayer.RData.zip](https://github.com/robjhyndman/forecast/files/1490625/autolayer.RData.zip)

**Comment by mitchelloharawild** [comment]
Thanks for reporting this, resolved in #620

```R
autoplot(trainning_series, series = "Data") + 
        autolayer(fc_1, series = "Forecast") +
        autolayer(fitted(fc_1), series = "Fitted")
```
![rplot](https://user-images.githubusercontent.com/16127127/33070984-e91ce7ec-cf0d-11e7-8788-29859818a17e.png)

**Comment by luiscarlosjr** [comment]
That's great, thank you!
Cheers!
